### PR TITLE
sequential algorithms examples

### PR DESCRIPTION
**Pull request by depombo**

### CODE REVIEW

File: src/advanced_examples.md [hunk 1]
```diff
@@ -484,49 +484,6 @@ on start fn {
...
-on start {
-  const count = [1, 2, 3, 4, 5]
-  const byTwos = count.map(fn (n: int64): int64 = n * 2)
-  count.map(fn (n: int64) = toString(n)).join(', ').print()
-  byTwos.map(fn (n: int64) = toString(n)).join(', ').print()
-  emit exit 0
-}
-```
-
-Alan does not allow arbitrary loops or recursion, but you can still loop over data. This example shows the primary way to do iteration in Alan using the functional [array api](./builtins/array_api.md) which can be parallelized by the AVM.
```
Comment (dfellis) [inline src/advanced_examples.md:501]: I think this one is still valid to keep. It's how we actually want people to manipulate arrays.

File: src/sequential_algorithms.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+on start {
+  const count = [1, 2, 3, 4, 5]
+  const byTwos = count.map(fn (n: int64): int64 = n * 2)
+  count.map(fn (n: int64) = toString(n)).join(', ').print()
+  byTwos.map(fn (n: int64) = toString(n)).join(', ').print()
+  emit exit 0
+}
+```
+This example shows the preferred way to do iteration in Alan using the functional [array api](./builtins/array_api.md) which can be parallelized by the AVM. The remaining examples can't be parallelized by the AVM, but they are guaranteed to terminate.
```
Comment (dfellis) [inline src/sequential_algorithms.md:18]: This doesn't belong here because it isn't explicitly sequential and may parallelize.

Comment (depombo) [reply src/sequential_algorithms.md:18]: hmm I see your point. I thought of a few ways to handle this, but I will put this back where it was and just link them

File: src/sequential_algorithms.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
+## Expressing Sequential Algorithms in Alan
```
Comment (dfellis) [inline src/sequential_algorithms.md:1]: Thinking about it. Should this simply be the `@std/seq` documentation in the standard library that we haven't written yet?

File: src/SUMMARY.md [hunk 4]
```diff
@@ -45,6 +45,7 @@
   - [@std/trig](./std_trig.md)
 - [Examples with Explanations](./examples_with_explanations.md)
   - [The many forms of "Hello, World!"](./many_forms_hello_world.md)
+  - [Sequential Algorithms](./sequential_algorithms.md)
```
Comment (dfellis) [inline src/SUMMARY.md:48]: This needs to be moved to the `@std/...` section and the new filename used.

File: src/std_seq.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
+## @std/seq to express Sequential Algorithms in Alan
+
+Alan does not allow arbitrary, or classical, loops or recursion and [the preferred way to do iteration is with arrays](./advanced_examples.md#loopln). However, there are algorithms people need to write that are inherently sequential, and often non-deterministic on the number of steps needed to take, such as with numeric approximation algorithms like Newton-Raphson, or anything written as a recursive function call. Most of the problems developers need to solve do not require this linearity, but for when it really is required, we are reintroducing this power in a controlled manner that still allows the AVM to be able to plan around these algorithms and provide the guarantee of termination. The syntax `@std/seq` provides was [designed (and only partially implemented so far)](https://github.com/alantech/alan/blob/main/rfcs/007%20-%20Sequential%20Algorithms%20RFC.md) to *not* feel like classical control flow in order to nudge developers to find better, more deterministic and more parallelizable mechanisms, instead.
```
Comment (dfellis) [inline src/std_seq.md:3]: # :clap:

File: src/std_seq.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+}
+```
+
+`doWhile` always runs at least once (unless the `seq` has reached its `limit` or it was constructed with an initial `limit` of `0`) and uses the return value of the function to determine if it should continue or not, so its `bodyFn` has the following signature:
+
+```alan
+fn bodyFn(): bool
+```
+
+### `recurse_fib.ln`
```
Comment (dfellis) [inline src/std_seq.md:111]: Maybe point out that this is not implemented in the AVM yet but only the js-runtime?

**Review by dfellis — APPROVED** [review]

**Pull request merged**